iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java

```diff
@@ -955,4 +1046,44 @@ private void validateBasicStats(Table icebergTable, String dbName, String tableN
       Assert.assertEquals(summary.get(entry.getValue()), hmsParams.get(entry.getKey()));
     }
   }
+
+  private void validateMigration(String tableName, int recordCount) {
+    List<Object[]> originalResult = shell.executeStatement("SELECT * FROM " + tableName + " ORDER BY a");
+    Assert.assertEquals(recordCount, originalResult.size());
+    List<Object[]> originalDescribe = shell.executeStatement("DESCRIBE FORMATTED " + tableName);
+    validateDescribeOutput(originalDescribe, fileFormat.name());
+    shell.executeStatement("ALTER TABLE " + tableName + " SET TBLPROPERTIES " +
+        "('storage_handler'='org.apache.iceberg.mr.hive.HiveIcebergStorageHandler')");
+    List<Object[]> alterResult = shell.executeStatement("SELECT * FROM " + tableName + " ORDER BY a");
+    Assert.assertEquals(originalResult.size(), alterResult.size());
+    List<Object[]> alterDescribe = shell.executeStatement("DESCRIBE FORMATTED " + tableName);
```

Comment:
Can we use `TestHiveMetastore#loadTable` instead and check the contents of the `sd`?

Reply:
Sure. Thanks for pointing this out.